src/main/java/task2/Calculator.java

```diff
@@ -0,0 +1,185 @@
+                if (!newStr.equals(expr))
+                    keysToDelete.add(kek.getKey());
+                expr = newStr;
+            }
+            System.out.println(expr);
+        }
+        System.out.println(var);
+        if (contains(expr, var)) {
+            return clear(expr, var);
+        }
+        return expr;
+    }
+
+    private static boolean contains(String str, Map<String, Object> var) {
+        boolean contains = false;
+        char chr = 'a';
+        for (char i : str.toCharArray()) {
+            if (i >= 'a' && i <= 'z') {
+                contains = true;
+                chr = i;
+                break;
+            }
+        }
+        if (var.get(String.valueOf(chr)) == null)
+            return false;
+        return contains;
+    }
+
+    private static ArrayList<String> tok(String expression) {
+        String expr = expression.replaceAll(" ", "");
+        String[] lets = expr.split("");
+        StringBuffer buf = new StringBuffer();
+        List<String> kek = new LinkedList<>();
+
+        for (int i = 0; i < lets.length; i++) {
+            if (i == 0 && lets[i].equals("-")) {
+                buf.append("-");
+            } else if (lets[i].equals("-") && Pattern.matches("[+*/()]", lets[i - 1])) {
+                buf.append("-");
+            } else if (Pattern.matches("[0-9]", lets[i])) {
+                buf.append(lets[i]);
+            } else if (Pattern.matches("[-+*/()]", lets[i])) {
+                if (buf.length() > 0) {
+                    kek.add(buf.toString());
+                    buf.delete(0, buf.length());
+                }
+                kek.add(lets[i]);
+            } else if (Pattern.matches("[a-z]", lets[i])) {
+                kek.add(lets[i]);
+            }
+            if (i == lets.length - 1 && buf.length() > 0) {
+                kek.add(buf.toString());
+                buf.delete(0, buf.length());
+            }
+        }
+
+        ArrayList<String> list = new ArrayList<>();
+        for (String i : kek)
+            list.add(i);
+        return list;
+    }
+
+    private static String[] post(ArrayList<String> toks) {
+        StringBuffer out = new StringBuffer();
+        Deque<String> stk = new ArrayDeque<>();
+
+        for (String token : toks) {
+            if (operators.containsKey(token)) {
+                while (!stk.isEmpty() && prec(token, stk.peek()))
+                    out.append(stk.pop()).append(' ');
+                stk.push(token);
+            } else if ("(".equals(token)) {
+                stk.push(token);
+            } else if (")".equals(token)) {
+                while (!"(".equals(stk.peek()))
+                    out.append(stk.pop()).append(' ');
+                stk.pop();
+            } else {
+                out.append(token).append(' ');
+            }
+        }
+
+        while (!stk.isEmpty())
+            out.append(stk.pop()).append(' ');
+
+        return out.toString().split(" ");
+    }
+
+    private enum op {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The enum name 'op' doesn't match '[A-Z][a-zA-Z0-9]*'](https://app.codacy.com/manual/thejerome/IFMO_Servlets_20190915/pullRequest?prid=5253422)